bodo/tests/test_snowflake_iceberg.py

```diff
@@ -94,7 +118,6 @@ def impl(table_name, conn, db_schema):
     )
 
 
-@temp_env_override({"AWS_REGION": "us-east-1"})
```

Comment:
No longer need these, cause Iceberg IO is PyArrow so it detects AWS_DEFAULT_REGION